pycloudlib/lxd/instance.py

```diff
@@ -371,6 +371,7 @@ def snapshot(self, snapshot_name):
             snapshot_name: name to call snapshot
         """
         self.clean()
+        self.shutdown()
```

Comment:
Hrm, this might be a prob for ephemeral instances right because the instance will disappear after stop? Maybe we don't want to call this when ephemeral VM?
```suggestion
        if not self._is_ephemeral:
            self.shutdown()
```

Reply:
The change is fine, but on UA, do you create snapshots like we do on cloud-init (note that I mean snapshot generally, not the lxd snapshot command)? Can you even take snapshots of an ephemeral VM?